protocol/messenger.go

```diff
@@ -460,19 +461,23 @@ func NewMessenger(
 	ctx, cancel := context.WithCancel(context.Background())
 
 	messenger = &Messenger{
-		config:                     &c,
-		node:                       node,
-		identity:                   identity,
-		persistence:                sqlitePersistence,
-		transport:                  transp,
-		encryptor:                  encryptionProtocol,
-		sender:                     sender,
-		anonMetricsClient:          anonMetricsClient,
-		anonMetricsServer:          anonMetricsServer,
-		telemetryClient:            telemetryClient,
-		pushNotificationClient:     pushNotificationClient,
-		pushNotificationServer:     pushNotificationServer,
-		communitiesManager:         communitiesManager,
+		config:                 &c,
+		node:                   node,
+		identity:               identity,
+		persistence:            sqlitePersistence,
+		transport:              transp,
+		encryptor:              encryptionProtocol,
+		sender:                 sender,
+		anonMetricsClient:      anonMetricsClient,
+		anonMetricsServer:      anonMetricsServer,
+		telemetryClient:        telemetryClient,
+		pushNotificationClient: pushNotificationClient,
+		pushNotificationServer: pushNotificationServer,
+		communitiesManager:     communitiesManager,
+		communitiesKeyDistributor: &CommunitiesKeyDistributorImpl{
```

Comment:
Just cosmetics, but we could have a `NewKeyDistributor()` factory fn for this

Reply:
Was thinking about factory as well, but since this struct doesn't need any special initialization, I left it as is.